source/extensions/filters/common/rbac/matchers.cc

```diff
@@ -157,6 +159,16 @@ bool PolicyMatcher::matches(const Network::Connection& connection,
          principals_.matches(connection, headers, metadata);
 }
 
+bool RequestedServerNameMatcher::matches(const Network::Connection& connection,
+                                         const Envoy::Http::HeaderMap&,
+                                         const envoy::api::v2::core::Metadata&) const {
+  absl::string_view requested_server_name = connection.requestedServerName();
+  if (!requested_server_name.empty()) { // test for null string_view
```

Comment:
That's because you only passing the data() to `StringMatcher::match`, but this is still wrong. string_view doesn't have to be null terminated, so the string you passed in to Matcher could be longer than the string_view.

Can you change StringMatcher::match to take `absl::string_view` instead? That would avoid a copy too.